### PR TITLE
For Code Review Submission

### PR DESCRIPTION
**Pull request by sutherm**
# Pull Request Template

## CI Checks
- [ ] The notebook runs from start to finish on all operating systems:
   - [ ] Mac
   - [ ] Windows
   - [ ] Linux 
### Reproducibility
- [ ] Are the data downloaded in the code
- [ ] Are paths created to ensure they work on all operating systems (using  os.path.join)
- [ ] Are comments used to clarify the contents of the code that can’t be clarified using expressive variable and function names alone? (not too many comments - just enough)
- [ ] Does the notebook run from start to finish?

### Functions
- [ ] Do functions follow PEP 8 format conventions?
- [ ] Are function docstrings clear (all  inputs and outputs clearly  described and defined)
- [ ] Are function names expressive (the name describes what the function does)?
- [ ] Are functions easy to understand and read?
- [ ] How many tasks does each function do? (ideally a function does one thing well).
### Package imports
- [ ] Are standard modules (those included with the base  python install) vs. third party (related but externally developed tools) import groups correct with appropriate spacing in between each group?
- [ ] Are variable names throughout the code, expressive?
- [ ] Suggest changes if not, highlight what was done well
- [ ] Is the code overall easy to understand and read? Are there things that would make it more clear / cleaner?

### DRY Code
- [ ] Are segments of code repeated in the file or is the code DRY?
- [ ] Are loops used to optimize DRY code?
- [ ] Are functions used to optimize DRY code?
- [ ] Are there any areas that could be potentially improved (you can suggest improvements OR you can just highlight parts of the code where you suspect it could  be cleaner / more efficient.

### Novel Approaches to Problem solving
- [ ] Highlight any novel approaches to completing the assignment.

### CODE REVIEW

File: sutherland_mike_ndvi.py [hunk 1]
```diff
@@ -0,0 +1,603 @@
...
+        band = rxr.open_rasterio(aband,
+                                 masked=True).rio.clip(clip_extent).squeeze()
+
+        if valid_drange:
+            mask = ((band < valid_drange[0]) | (band > valid_drange[1]))
+            cleaned_band = band.where(~xr.where(mask, True, False))
+
+        proc_bands.append(cleaned_band)
+
+    ndvi_full = (proc_bands[1]-proc_bands[0]) /         (proc_bands[1]+proc_bands[0])
```
Comment (nlamkey) [inline sutherland_mike_ndvi.py:285]: There is some empty space after the / symbol.

File: sutherland_mike_ndvi.py [hunk 2]
```diff
@@ -0,0 +1,603 @@
...
+    formatted_df : pandas dataframe
+        A formatted datafrmae containing requisite column names and indexed
+        by acqusition date.
+    """
+    
+    formatted_df = pd.DataFrame(inlist, columns=["date", "site", "mean_ndvi"])
+    formatted_df["date"] = pd.to_datetime(formatted_df["date"])
+    formatted_df.set_index("date", inplace=True)
+    return formatted_df
+
```
Comment (nlamkey) [inline sutherland_mike_ndvi.py:319]: You're functions look really great!

File: sutherland_mike_ndvi.py [hunk 3]
```diff
@@ -0,0 +1,603 @@
...
+# Empty list creation; needed for subsequent processing steps
+df_list = []
+
+# Extract & combine acq date, location and NDVI information into a pandas df
+loc_date = fetch_site_date(bands_dpath)
+ndvi_calc = create_ndvi(bands_dpath, crop_extent, vals, valid_pixels)
+df_list.append(loc_date + ndvi_calc)
+final_df = list_to_df(df_list)
+final_df
+
```
Comment (nlamkey) [inline sutherland_mike_ndvi.py:349]: Your code looks really clean, not much else to say here.

File: sutherland_mike_ndvi.py [hunk 4]
```diff
@@ -0,0 +1,603 @@
...
+    dirs = sorted(glob(os.path.join(site, "landsat-crop", "*")))
+    for dir in dirs:
+        loc_date = fetch_site_date(dir)
+        ndvi_calc = create_ndvi(dir, crop_extent, vals, valid_pixels)
+        ndvi_df_list.append(loc_date + ndvi_calc)
+
+# Final pandas df containing information for all Landsat scenes
+final_df = list_to_df(ndvi_df_list)
+final_df
+
```
Comment (nlamkey) [inline sutherland_mike_ndvi.py:441]: Again your code looks great.

File: sutherland_mike_ndvi.py [hunk 5]
```diff
@@ -0,0 +1,603 @@
...
+
+# Plotting initialization
+f, ax = plt.subplots(figsize=(12, 8))
+
+# Drop nan values in df in order to correctly "connect the dots" in the plot
+final_df.dropna(inplace=True)
+
+# Plotting the data according to each respective site location
+for title, group in final_df.groupby("site"):
```
Comment (nlamkey) [inline sutherland_mike_ndvi.py:511]: I like the loop you used for your plot. I don't think I thought to do this in my code.

File: sutherland_mike_ndvi.py [hunk 6]
```diff
@@ -0,0 +1,603 @@
...
+    
+    site_id = os.path.normpath(inpath).split(os.sep)[-3]
+    date_id = os.path.basename(os.path.normpath(inpath))[10:18]
+    date_site = [date_id, site_id]
+    return date_site
+
+
+def create_ndvi(rastpath, clip_shp, cloud_vals,
+                valid_drange=None):
+    """Iterate through a red, NIR bands in each directory and open as an
```
Comment (nlamkey) [inline sutherland_mike_ndvi.py:240]: @sutherm Could summerazie input and output a little better. Like, Takes a path to tif files, clips to an extent, filters for clouds and valid range, calculates an ndvi and mean ndvi.

**Review by nlamkey — COMMENTED** [review]
I wasn't able to check off the boxes for some reason but I went through the list and indicated comments from the list. I think overall the workflow was clean and easy to follow.

**Comment by sutherm** [comment]
@nlamkey thank you for the code review; I've reviewed the comments and will be mindful of adding more descriptive language/comments to improve clarity in future coding projects

**Pull request closed without merging**